packages/berry-pnpify/sources/generateSdk.ts

```diff
@@ -2,47 +2,73 @@ import {Filename, NodeFS, PortablePath, xfs, ppath} from '@berry/fslib';
-  const tsserver = ppath.join(tssdk, `lib/tsserver.js` as PortablePath);
+  const data = JSON.parse(content);
+  const patched = `${JSON.stringify({...data, ...settings}, null, 2)}\n`;
+
+  await xfs.mkdirpPromise(ppath.dirname(settingsPath));
+  await xfs.changeFilePromise(settingsPath, patched);
+}
+
+const generateTypescriptWrapper = async (projectRoot: PortablePath, tssdk: PortablePath) => {
+  const typescript = ppath.join(tssdk, `typescript` as PortablePath);
+  const manifest = ppath.join(typescript, `package.json` as PortablePath);
+  const tsserver = ppath.join(typescript, `lib/tsserver.js` as PortablePath);
 
   const relPnpApiPath = ppath.relative(ppath.dirname(tsserver), ppath.join(projectRoot, `.pnp.js` as Filename));
 
-  await xfs.removePromise(tssdk);
   await xfs.mkdirpPromise(ppath.dirname(tsserver));
-  await xfs.writeFilePromise(tssdkManifest, JSON.stringify({name: 'typescript', version: `${dynamicRequire('typescript/package.json').version}-pnpify`}, null, 2));
-  await xfs.writeFilePromise(tsserver, TEMPLATE(relPnpApiPath));
+  await xfs.writeFilePromise(manifest, JSON.stringify({name: 'typescript', version: `${dynamicRequire('typescript/package.json').version}-pnpify`}, null, 2));
+  await xfs.writeFilePromise(tsserver, TEMPLATE(relPnpApiPath, "typescript/lib/tsserver", true));
 
-  const settings = ppath.join(projectRoot, `.vscode/settings.json` as PortablePath);
-  const content = await xfs.existsPromise(settings) ? await xfs.readFilePromise(settings, `utf8`) : `{}`;
+  await addVSCodeWorkspaceSettings(projectRoot, {'typescript.tsdk': NodeFS.fromPortablePath(ppath.relative(projectRoot, ppath.dirname(tsserver)))});
+}
 
-  const data = JSON.parse(content);
-  data[`typescript.tsdk`] = NodeFS.fromPortablePath(ppath.relative(projectRoot, ppath.dirname(tsserver)));
-  const patched = `${JSON.stringify(data, null, 2)}\n`;
+export const generateEslintWrapper = async (projectRoot: PortablePath, tssdk: PortablePath) => {
+  const eslint = ppath.join(tssdk, `eslint` as PortablePath);
+  const manifest = ppath.join(eslint, `package.json` as PortablePath);
+  const api = ppath.join(eslint, `lib/api.js` as PortablePath);
+
+  const relPnpApiPath = ppath.relative(ppath.dirname(api), ppath.join(projectRoot, `.pnp.js` as Filename));
+
+  await xfs.mkdirpPromise(ppath.dirname(api));
+  await xfs.writeFilePromise(manifest, JSON.stringify({name: 'eslint', version: `${dynamicRequire('eslint/package.json').version}-pnpify`, main: 'lib/api.js'}, null, 2));
+  await xfs.writeFilePromise(api, TEMPLATE(relPnpApiPath, "eslint", false));
+
+  await addVSCodeWorkspaceSettings(projectRoot, {'eslint.nodePath': NodeFS.fromPortablePath(ppath.relative(projectRoot, ppath.dirname(eslint)))});
+}
+
+export const generateSdk = async (projectRoot: PortablePath, targetFolder: PortablePath | null): Promise<any> => {
+  if (targetFolder === null)
+    targetFolder = projectRoot;
+
+  const tssdk = ppath.join(targetFolder, `tssdk` as PortablePath);
```

Comment:
Since it's no longer the "typescript sdk", maybe rename it to something else? And since it's vscode specific we could just store it within this directory and remove the `targetFolder` option altogether, what do you think?

```suggestion
  const target = ppath.join(projectFolder, `.vscode/pnpify` as PortablePath);
```